lib/matplotlib/tests/test_colorbar.py

```diff
@@ -712,6 +713,20 @@ def test_colorbar_label():
     assert cbar3.ax.get_xlabel() == 'horizontal cbar'
 
 
+@image_comparison(['colorbar_keeping_xlabel.png'], style='mpl20')
+def test_keeping_xlabel():
+    # github issue #23398 - xlabels being ignored in colorbar axis
+    np.random.seed(seed=19970527)
+    arr = np.random.rand(10, 10)
+    fig, axes = plt.subplots()
+    im = axes.imshow(arr)
+    divider = make_axes_locatable(axes)
+    cbar_ax = divider.append_axes('right', size='5%', pad=0.5)
+    cbar = plt.colorbar(im, cax=cbar_ax)
```

Comment:
just use the default colorbar:
```suggestion
    cbar = plt.colorbar(im)
```